src/entries/popup/pages/home/Points/ClaimOverview.tsx

```diff
@@ -0,0 +1,403 @@
+import { useWalletName } from '~/entries/popup/hooks/useWalletName';
+import { ROUTES } from '~/entries/popup/urls';
+import { zIndexes } from '~/entries/popup/utils/zIndexes';
+
+import * as wallet from '../../../handlers/wallet';
+
+import ConsoleText from './ConsoleText';
+import { CLAIM_MOCK_DATA } from './references';
+import { invalidatePointsQuery, refetchPointsQuery } from './usePoints';
+import { RAINBOW_TEXT, getDelayForRow } from './utils';
+
+export function ClaimOverview() {
+  const { state } = useLocation();
+  const claimNetwork = state?.claimNetwork as ChainId;
+  const claimAmount: string = state?.claimAmount;
+  const [bridgeSuccess, setBridgeSuccess] = useState(false);
+  const [bridgeError, setBridgeError] = useState<string | undefined>();
+  const [waitToDisplay, setWaitToDisplay] = useState(true);
+  const [showSummary, setShowSummary] = useState(false);
+  const [error, setError] = useState<PointsErrorType | undefined>();
+  const navigate = useRainbowNavigate();
+  const { currentAddress } = useCurrentAddressStore();
+  const { currentCurrency: currency } = useCurrentCurrencyStore();
+  const { displayName } = useWalletName({ address: currentAddress });
+
+  const { currentAddress: address } = useCurrentAddressStore();
+
+  const eth = useNativeAsset({ chainId: ChainId.mainnet });
+  const ethPrice = eth?.nativeAsset?.price?.value;
+  const claimableBalance = convertRawAmountToBalance(claimAmount, {
+    decimals: 18,
+    symbol: eth?.nativeAsset?.symbol,
+  });
+  const claimablePriceDisplay = convertAmountAndPriceToNativeDisplay(
+    claimableBalance.amount,
+    ethPrice || '0',
+    currency,
+  );
+
+  const opEth = useNativeAssetForNetwork({ chainId: ChainId.optimism });
+  const destinationEth = useNativeAssetForNetwork({ chainId: ChainId.base });
+
+  const { data: swapSlippage } = useSwapSlippage({
+    chainId: ChainId.optimism,
+    toChainId: destinationEth?.chainId || ChainId.mainnet,
+    sellTokenAddress: opEth?.address as Address,
+    buyTokenAddress: destinationEth?.address as Address,
+    sellAmount: claimableBalance.amount || 0,
+    buyAmount: '',
+  });
+
+  const { mutate: claimRewards, isSuccess: claimSuccess } = useMutation<
+    ClaimUserRewardsMutation['claimUserRewards']
+  >({
+    mutationFn: async () => {
+      const response =
+        process.env.IS_TESTING === 'true'
+          ? CLAIM_MOCK_DATA
+          : await metadataPostClient.claimUserRewards({ address });
+      const claimInfo = response?.claimUserRewards;
+
+      if (claimInfo?.error) {
+        setError(claimInfo?.error.type);
+      }
+
+      // clear and refresh claim data so available claim UI disappears
+      invalidatePointsQuery(address);
+      refetchPointsQuery(address);
+
+      return claimInfo;
+    },
+    onSuccess: async (d) => {
+      if (
+        claimNetwork !== ChainId.optimism &&
+        d?.txHash &&
+        opEth &&
+        destinationEth
+      ) {
```

Comment:
is there any option that this if happens? for example, if  `claimNetwork !== ChainId.optimism` should we execute the mutationFn ? feels like we should be using this if condition to execute the fn not after is executed (potentially using wrong params)

Reply:
discussed offline, added a clarifying comment